src/transformers/models/audio_spectrogram_transformer/modeling_audio_spectrogram_transformer.py

```diff
@@ -267,15 +306,21 @@ def forward(self, hidden_states: torch.Tensor, input_tensor: torch.Tensor) -> to
         return hidden_states
 
 
-# Copied from transformers.models.vit.modeling_vit.ViTLayer with ViT->AST
+AST_ATTENTION_CLASSES = {
+    "eager": ASTAttention,
+    "sdpa": ASTSdpaAttention,
```

Comment:
```suggestion
    "sdpa": ASTSdpaSelfAttention,
```
alright from what I got, if the two classes are the same then we should be able to do that no?

Reply:
They are not quite the same, ASTSdpaAttention is extending ASTAttention with a function for pruning, like in ViT implementation.